OpenTrons/opentrons-api
# feat(protocol-designer, step-generation, components): pipette edit, moveLabware, ui changes

closes RAUT-512 RAUT-519 RAUT-505 RAUT-509 RAUT-523

# Overview

Various things outlined in Changelog below. Basically fixing up the UI for the Create File Wizard, modifying tiprack and pipette selections, adding back custom tiprack support and adding some timeline errors to `moveLabware` when moving a labware out of a TC or H-S when the lid/latch is closed/is shaking. Additionally, I added an experimental setting to allow users the option to select any tiprack for their pipette.

# Test Plan

Should smoke test this. Focus on these sections:
1. create file wizard - look at the tiprack section, add a custom tiprack, etc. 
2. edit pipette section - make sure the pipettes are correct for the flex
3. `moveLabware` - make sure errors show up when expected and the dropdowns work/display correct info and `offDeck` slot option is appearing as expected
4. toggle on the "Allow all tiprack" setting and make sure you can select which ever tiprack you want

# Changelog

- fixes devTools warnings throughout `CreateFileWizard` by adding some `React.useEffects` to update formik
- allow the Flex module edit section to only allow users to select GEN2 TC and GEN2 Temp, rather than allowing the GEN1 options as well
- update the Pipette edit section to show correct pipettes for each robot and show only default tipracks.
- update the `CreateFileWizard` tiprack sections to have custom tiprack support
- various UI changes to the `CreateFileWizard` to make it look nicer with consistent spacing/font size
- adjust the OT-2 deck map view to display the map + module tags clearer
- add the LED lighting to the temperature and H-s modules upon heating and cooling
- changes to the `moveLabwareForm`'s location dropdown so the OT-2 TC says "slot 7, 8, 10, 11" instead of the `slot_7_8_10_11` string and allow an `offDeck` location only when it is a manual movement
- create a `errorCreator`: `heaterShakeLatchClosed` and add logic to `moveLabware` so you get a timeline error if you try to move a labware out of the H-S or TC when the latch/lid is closed or the H-S is shaking
- added an advanced setting under `Experimental Settings` for allowing all tiprack selection. When toggled on, and on an OT-2, users can select any OT-2 tip + 3rd party tip in our labware library (like TipOne). For a Flex, users can select any of the flex tipracks

# Review requests

see test plan

# Risk assessment

low
